tools/gen-providers.ts

```diff
@@ -3638,16 +3770,47 @@ function generateAws(gen: Generator) {
   gen.generateParams(sqs_queue);
   gen.generateParams(sqs_queue_policy);
   gen.generateParams(lb);
+  gen.generateDataSource(
+    'Datasource for lb',
+    'https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/lb',
+    lb,
+    [
+      stringAliasAttr('arn', "Arn", "AT.Arn"),
+      stringAttr("name"),
+      // "tags"
+    ]
+  );
```

Comment:
I think it didn't quite get the `lb` data source stuff to work - right?

It's because the params block for a `resource` isn't the same as the params block for a `data` of the same name.  
https://registry.terraform.io/providers/hashicorp/aws/2.70.0/docs/resources/lb
vs
https://registry.terraform.io/providers/hashicorp/aws/2.70.0/docs/data-sources/lb

Perhaps following the example of the ami data source lead you astray because it *only* has the param block for the data source https://registry.terraform.io/providers/hashicorp/aws/2.70.0/docs/data-sources/ami whereas almost every other type we have the resource block only, not the data block.

Reply:
Removed from history.